### PR TITLE
Added Amazon price tracker

### LINKED ISSUE #663: Amazon Price Tracker using JS
**Is your feature request related to a problem? Please describe.**
There are multiple scripts in Python more suited for Javascript, this being one. Having scripts in multiple languages for similar problems makes a whole lot of sense and is definitely useful.

**Describe the solution you'd like**
An amazon scrapper/tracker for products.

### PR DESCRIPTION
**Pull request by haarsh01**
# Description

Receive an email notification when a specific Amazon product has dropped below the desired price.
I used Nodemailer Module. It is a module for Node.js applications to allow easy as cake email sending.

Fixes #663

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Can you add a demo Video/GIF to the PR description? Thanks!

**Comment by haarsh01** [comment]
> Can you add a demo Video/GIF to the PR description? Thanks!

Hey, my pc is not responding to the screen recording I tried. Is this is compulsory?

**Comment by vybhav72954** [comment]
> > Can you add a demo Video/GIF to the PR description? Thanks!
> 
> Hey, my pc is not responding to the screen recording I tried. Is this is compulsory?

Try OBS or Bandicam. A demonstration helps us to ensure that such scripts work properly. alternatively send us some screenshot that can properly justify the working of scirpt

**Comment by HarshCasper** [comment]
> > Can you add a demo Video/GIF to the PR description? Thanks!
> 
> Hey, my pc is not responding to the screen recording I tried. Is this is compulsory?

Try using the `Awesome Screenshot` extension. Otherwise trying out locally would require wait for a few days.

**Comment by haarsh01** [comment]
> > > Can you add a demo Video/GIF to the PR description? Thanks!
> > 
> > 
> > Hey, my pc is not responding to the screen recording I tried. Is this is compulsory?
> 
> Try using the `Awesome Screenshot` extension. Otherwise trying out locally would require wait for a few days.

Ok got it. I will do it

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
@haarsh01 A slight miscommunication, we wanted that video in PR description, not in the codebase.

4 changes:


- The script is lacking README, we **need** a README file, check [template](https://github.com/HarshCasper/Rotten-Scripts/blob/master/TEMPLATE_README.md).
- Every script has a README in the directory, pls take inspiration from them. Your output image/test case needs to be on README as well.
- Remove the JPG file, Upload it on something like `imgur` or other similar files. Paste the link on README.
- Remove the mp4 file. Upload it on drive or similar platforms. Provide us with the link here (change the permission accordingly). If you want, you can paste it on README as well (not mandatory).

File: JavaScript/Amazon_Price_Tracker/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (vybhav72954) [inline JavaScript/Amazon_Price_Tracker/ReadMe.md:1]: Thabk you for adding the README, although there are multiple problems with the file.

A detailed list of [warnings](https://github.com/HarshCasper/Rotten-Scripts/runs/2335505295?check_suite_focus=true)

You can either use this list and/or use the [markdown guide](https://github.com/HarshCasper/Rotten-Scripts/blob/master/STYLE_GUIDELINES.md),

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: JavaScript/Amazon_Price_Tracker/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+const watchPrice = (priceTaget, url, schedule = '*/5 * * * * *') => {
+    cron.schedule(schedule, () => fetchPrice(url, priceTaget));
+};
+
+
+watchPrice( 53.99,
+    'https://www.amazon.com/Toshiba-HDTB420XK3AA-Canvio-Portable-External/dp/B079D3D8NR/ref=sr_1_6?dchild=1&keywords=hardisk&qid=1617019368&sr=8-6',
+
```
Comment (sohamsshah) [inline JavaScript/Amazon_Price_Tracker/index.js:46]: Remove extra lines. Probably use a simple linter or code formatter

File: package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (sohamsshah) [inline package.json:1]: Any specific reason why adding a `package.json` for Rotten Scripts?
`package.json` file is created when doing npm init or similar operations. Rotten Scripts isn't a JS npm package or a node app, thus please consider removing this file, it is not required.

**Review by sohamsshah — CHANGES_REQUESTED** [review]
The `index.js` file has some really good practices implemented. Appreciate the good work 👍 

Though have suggested some minor changes, after that it is good to merge!

File: package.json (hunk 3 above)
Comment (haarsh01) [reply package.json:1]: Ok, I will make the necessary changes.

**Review by vybhav72954 — APPROVED** [review]
@HarshCasper This looks good to me
@sohamsshah A final Review Please

**Comment by sohamsshah** [comment]
Hi @haarsh01 I speculate that you have removed `package.json` file itself from the folder. Then how would anyone could do `npm install` without it? That is why, `package.json` file is must for node.js based projects. 

Actually I meant to remove the `package.json` in the main root of the repo, which you have added and not the one which is for your project. Yotu can safely remove that from the root repo - [this one](https://github.com/haarsh01/Rotten-Scripts/blob/master/package.json) :) 
I hope you get my point. Otherwise, it looks good to me.
Thanks for working on this!

**Comment by haarsh01** [comment]
> Hi @haarsh01 I speculate that you have removed `package.json` file itself from the folder. Then how would anyone could do `npm install` without it? That is why, `package.json` file is must for node.js based projects.
> 
> Actually I meant to remove the `package.json` in the main root of the repo, which you have added and not the one which is for your project. Yotu can safely remove that from the root repo - [this one](https://github.com/haarsh01/Rotten-Scripts/blob/master/package.json) :)
> I hope you get my point. Otherwise, it looks good to me.
> Thanks for working on this!

Hey , I got your point . So should I add package.json again into the project and delete one from the root. Right?

**Comment by vybhav72954** [comment]
> > Hi @haarsh01 I speculate that you have removed `package.json` file itself from the folder. Then how would anyone could do `npm install` without it? That is why, `package.json` file is must for node.js based projects.
> > Actually I meant to remove the `package.json` in the main root of the repo, which you have added and not the one which is for your project. Yotu can safely remove that from the root repo - [this one](https://github.com/haarsh01/Rotten-Scripts/blob/master/package.json) :)
> > I hope you get my point. Otherwise, it looks good to me.
> > Thanks for working on this!
> 
> Hey , I got your point . So should I add package.json again into the project and delete one from the root. Right? @sohamsshah

Yes, Remove the npm package file from the root and add the required package file in your script directoy

**Comment by haarsh01** [comment]
@vybhav72954 Hey , done all the necessary changes

**Review by sohamsshah — APPROVED** [review]
LGTM @HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**